server/src/main/java/org/elasticsearch/monitor/fs/FsHealthService.java

```diff
@@ -30,32 +30,26 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
-import java.util.Set;
 import java.util.function.LongSupplier;
-import java.util.stream.Collectors;
 
 import static org.elasticsearch.monitor.StatusInfo.Status.HEALTHY;
 import static org.elasticsearch.monitor.StatusInfo.Status.UNHEALTHY;
 
 /**
- * Runs periodically and attempts to create a temp file to see if the filesystem is writable. If not then it marks the
- * path as unhealthy.
+ * Runs periodically and attempts to create a temp file to see if the filesystem is writable. If not then it marks the path as unhealthy.
  */
 public class FsHealthService extends AbstractLifecycleComponent implements NodeHealthService {
 
     private static final Logger logger = LogManager.getLogger(FsHealthService.class);
     private final ThreadPool threadPool;
+    private volatile StatusInfo statusInfo = new StatusInfo(HEALTHY, "not started");
```

Comment:
Updating the status by writing to those two `volatile` fields was not quite correct: you could in theory observe a healthy state when transitioning from broken to unhealthy. It looks to be fixable by ordering the reads and writes carefully but the reasoning is quite fragile and it's much simpler just to do this.